CRAYSAT-1935: WAR for `sat bootprep` image customization error (#5553)
Cray-HPE/docs-csm: troubleshooting/known_issues/sat_bootprep_image_customization_error.md

# Known Issue: `sat bootprep` image customization Error

> On CSM 1.5.2 systems, the following error occurs when you try to customize the image using the `sat bootprep` command directly from management nodes after cray-sat version 2.25.11.

Reviewer: Suggested simplified wording here:

```suggestion
On CSM 1.5.2 systems, the following error occurs when you try to customize an image using the `sat
bootprep` command directly from management nodes:
```

> On CSM 1.5.2 systems, the following error occurs when you try to customize the image using the `sat bootprep` command directly from management nodes after cray-sat version 2.25.11.

Reviewer: Suggested simplified wording here:

```suggestion
On CSM 1.5.2 systems, the following error occurs when you try to customize an image using the `sat
bootprep` command directly from management nodes:
```

    created_images = create_images(instance, args, ims_client)
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootprep/image.py", line 520, in create_images
    waiter.wait_for_completion()
  File "/sat/venv/lib/python3.9/site-packages/sat/waiting.py", line 331, in wait_for_completion
    super().wait_for_completion()
  File "/sat/venv/lib/python3.9/site-packages/sat/waiting.py", line 169, in wait_for_completion
    self._wait_polling_loop()
  File "/sat/venv/lib/python3.9/site-packages/sat/waiting.py", line 549, in _wait_polling_loop
    if self.member_has_completed(member):
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootprep/image.py", line 407, in member_has_completed
    if member.has_completed:
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootprep/input/image.py", line 513, in has_completed
    self.begin_image_configure()
  File "/sat/venv/lib/python3.9/site-packages/sat/cli/bootprep/input/image.py", line 405, in begin_image_configure
    self.image_configure_session = self.cfs_client.create_image_customization_session(
TypeError: create_image_customization_session() missing 1 required positional argument: 'image_name' 
```

> 

Reviewer: Then I think we can add the additional detail about which versions of the `sat` CLI (i.e. `cray-sat` image) and the SAT product stream are affected. Most likely, people are only here looking at this if they've encountered the issue. The problem was introduced in `sat` version 3.25.14 and fixed in 3.25.16.

```suggestion

This affects versions 3.25.14 and 3.25.15 of the `sat` CLI. The version of the `sat` CLI can be
viewed with `sat --version`. These versions of the `sat` CLI were included in SAT product versions
2.6.18 and 2.6.19, respectively.
```

## Workaround

Set the environment variable `SAT_IMAGE` to point at the older version of
`cray-sat` provided by CSM 1.5.2 instead of the version from the newer SAT product stream.

> This will be recommended doing this only for `sat bootprep` commands. For example,

Reviewer: minor wording suggestion:

```suggestion
Do this only for `sat bootprep` commands. For example,
```

> # Known Issue: `sat bootprep` image customization Error

Reviewer: ```suggestion
# Known Issue: `sat bootprep` image customization error

```

> # Known Issue: `sat bootprep` image customization error

Reviewer 2: Remove the leading "Known issue:" to match consistency with other known issue headers. This also makes the table-of-content entry match the level-1 header.
```suggestion
# `sat bootprep` image customization error

```

Author: Addressed
